### PR TITLE
convert sign in to spa guide to single page

### PR DESCRIPTION
**Pull request by chrisdmills-okta**
## Description:
- **What's changed?** https://developer.okta.com/docs/guides/sign-into-spa/angular/next-steps/ converted to single page guide
- **Is this PR related to a Monolith release?** No

### Resolves:

* [OKTA-438730](https://oktainc.atlassian.net/browse/OKTA-438730)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+Install the SDK and add it as a dependency to your project:
+
+<StackSelector snippet="installsdk" noSelector />
+
+## Configure the SDK
+
+You need two values from the Okta application and the Admin Console that you worked with in [Create an Okta application](#create-an-okta-app-integration):
+
+* **Client ID** &mdash; find it in the applications list or on the application's **General** tab.
+* **Okta domain** &mdash; you can find the Okta Domain in the Admin Console's global header in the upper-right corner of the page. Click the section that displays your email and company name.  A drop-down box appears and displays general org information including the full Okta domain (for example, subdomain.okta.com).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md:108]:
```suggestion
* **Okta domain** &mdash; find the Okta Domain in the Admin Console's global header in the upper-right corner of the page. Click the section that displays your email and company name. A drop-down box appears and displays general org information including the full Okta domain (for example, `subdomain.okta.com`).
```

File: packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+In your application, you want to provide a button to sign the user in. When the user clicks this button, they are redirected to the Okta-hosted sign-in page so they can authenticate.
+
+> **Note:** To customize this sign-in page, see the [Style the Widget](/docs/guides/style-the-widget/style-okta-hosted/).
+
+<StackSelector snippet="login-redirect" noSelector />
+
+## Handle the callback from Okta
+
+After Okta authenticates a user, they're redirected back to your application via the callback route you [Define a callback route](#define-a-callback-route). When Okta redirects back, the URL fragment (the portion after `#`) contains either tokens for the user or an error if something went wrong.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md:122]:
```suggestion
After Okta authenticates a user, they're redirected back to your application through the callback route that you [define](#define-a-callback-route). When Okta redirects back, the URL fragment (the portion after `#`) contains either tokens for the user or an error if something went wrong.
```

File: packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+Your application must parse this information, and if tokens are present, store the user's tokens. Our SDK does this for you.
+
+<StackSelector snippet="handle-callback" noSelector />
+
+## Require authentication
+
+In many applications, you want to prevent the user from accessing certain routes or sections unless they are signed in. You can require authentication across the entire app or just require it for particular routes or controllers. Any route that doesn't specifically require authentication is accessible without signing in (also called anonymous access).
+
+It's important to note that protecting routes in your SPA does not truly prevent the user from accessing those parts of your application. After all, it's JavaScript running in the browser and anyone could open the browser's developer tools and change things! Protecting routes provides a consistent and good experience for your users. The real security enforcement must be done in the API that your single-page app calls (see [Use the access token](#use-the-access-token)). See also the [Protect your API endpoints guide](/docs/guides/protect-your-api/).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md:132]:
```suggestion
It's important to note that protecting routes in your SPA doesn't truly prevent the user from accessing those parts of your application. After all, it's JavaScript running in the browser, and anyone could open the browser's developer tools and change things! Protecting routes provides a consistent and good experience for your users. The real security enforcement must be done in the API that your single-page app calls (see [Use the access token](#use-the-access-token)). Also, see the [Protect your API endpoints guide](/docs/guides/protect-your-api/).
```

File: packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+## Get info about the user
+
+When a user signs in, their profile information (stored in Okta) is made available to your application. It's common to use this info to update your app's UI.
+
+By default, the profile items (called "claims") that Okta returns include the user's email address, first name, and last name. 
+
+<StackSelector snippet="getuserinfo" noSelector />
+
+<!-- You can also customize the items (called claims) that are returned from Okta. See [Token customization guide]. -->
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md:154]: why is this commented out? there are numerous customize token guides, starting with this one: https://developer.okta.com/docs/guides/customize-tokens-returned-from-okta/overview/

**Review by susanharper-okta — COMMENTED** [review]
couple things

File: packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+**Learning outcomes**
+
+* Set up an Okta app integration to represent your SPA inside your Okta org.
+* Add dependencies to your app and add authentication functionality to it.
+* Sign a user in to your application.
+
+**What you need**
+
+* An Okta Developer Edition organization. Don't have one? [Create one for free](https://developer.okta.com/signup).
```
[the remark this answers is not in the document]
Comment (chrisdmills-okta) [reply packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md:21]: fixed

File: packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+* Basic knowledge of building JavaScript applications.
```
[the remark this answers is not in the document]
Comment (chrisdmills-okta) [reply packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md:22]: fixed

File: packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+* A project or application that you want to add authentication to.
```
[the remark this answers is not in the document]
Comment (chrisdmills-okta) [reply packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md:23]: fixed

File: packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+
+See the [OAuth 2.0 for Browser-Based Apps specification](https://tools.ietf.org/html/draft-ietf-oauth-browser-based-apps-05#page-10) for the latest spec information on using refresh tokens with browser-based apps.
+
+## Define a callback route
+
+To sign users in, your application redirects the browser to an Okta-hosted sign-in page. Okta then redirects back to your application with information about the user. You can learn more about how this works on [Okta-hosted flows](/docs/concepts/okta-hosted-flows/).
+
+The first thing that you need to define is how Okta calls your app after a user is authenticated. This is called a callback route or redirect URI.
+
+The callback route is not seen by the user, and it's not the user's final destination. It's just one step in the authentication redirect flow.
```
[the remark this answers is not in the document]
Comment (chrisdmills-okta) [reply packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md:50]: Updated

File: packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+
+1. Sign in to your Okta organization with your administrator account.
+
+    <a href="https://developer.okta.com/login" target="_blank" class="Button--blue">Go to Admin Console</a>
+
+1. In the Admin Console, go to **Applications** > **Applications**.
+1. Click **Create App Integration**.
+1. Select **OIDC - OpenID Connect** as the **Sign-in method**.
+1. Select **Single-Page Application** as the **Application type** and click **Next**.
+    > **Note:** It is important to choose the appropriate application type for apps that are public clients. Failing to do so may result in Okta API endpoints attempting to verify an app's client secret, which public clients are not designed to have, and would break the sign-in or sign-out flow.
```
[the remark this answers is not in the document]
Comment (chrisdmills-okta) [reply packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md:68]: Updated

File: packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+1. Enter a name for your app integration (or leave the default value).
+1. Select **Authorization Code** and **Refresh Token** as the **Grant type**. This enables Authorization Code flow with PKCE for your application and the ability to refresh the access token when it expires without prompting the user to re-authenticate.
+1. Enter values for the **Sign-in redirect URIs**. This is the callback from [Define a callback route](#define-a-callback-route). Add values for local development (for example, `http://localhost:8080/login/callback` or `http://localhost:8080/authentication/login-callback` for Blazor) and production (for example, `https://app.example.com/login/callback`).
+
+    If your OpenID Connect client has multiple redirect URIs and you want to use a single redirect URI with a wildcard for the subdomain, select the **Allow wildcard * in sign-in redirect URI** checkbox.
```
[the remark this answers is not in the document]
Comment (chrisdmills-okta) [reply packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md:73]: Added

File: packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md (hunk 1 above)
Comment (chrisdmills-okta) [reply packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md:108]: Updated!

File: packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md (hunk 2 above)
Comment (chrisdmills-okta) [reply packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md:122]: Updated.

File: packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md (hunk 3 above)
Comment (chrisdmills-okta) [reply packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md:132]: Updated!

File: packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md (hunk 4 above)
Comment (chrisdmills-okta) [reply packages/@okta/vuepress-site/docs/guides/sign-into-spa/main/index.md:154]: I really have no idea ;-) 

But I've uncommented it now, and added in a proper link.

**Comment by chrisdmills-okta** [comment]
Thanks a lot @susanharper-okta for the great comments. I've fixed all of those, and now I just need to fix the merge conflicts.

**Comment by chrisdmills-okta** [comment]
OK, merge conflicts fixed @susanharper-okta . *phew*

**Review by susanharper-okta — APPROVED** [review]
LGTM

**Pull request merged**